### PR TITLE
refactor to use hooks

### PR DESCRIPTION
**Pull request by Gracetemitope**
In this branch, I refactor the code to use hooks(useState)

### CODE REVIEW

File: src/Component/Calculator.js [hunk 1]
```diff
@@ -37,39 +35,38 @@ class Calculator extends React.Component {
         </div>
         <div className="calculator d-flex flex-column">
           <div className="calc-row d-flex align-content-stretch">
-            <button className="p-5" onClick={() => { this.eventHandler(calculate(this.state, 'AC')); }}>AC</button>
-            <button className="p-5" onClick={() => { this.eventHandler(calculate(this.state, '+/-')); }}>+/-</button>
-            <button className="p-5" onClick={() => { this.eventHandler(calculate(this.state, '%')); }}>%</button>
-            <button className="p-5 orange" onClick={() => { this.eventHandler(calculate(this.state, '÷')); }}>÷
+            <button className="p-5" onClick={() => { eventHandler(calculate(obj, 'AC')); }}>AC</button>
+            <button className="p-5" onClick={() => { eventHandler(calculate(obj, '+/-')); }}>+/-</button>
+            <button className="p-5" onClick={() => { eventHandler(calculate(obj, '%')); }}>%</button>
```
Comment (DeliceLydia) [inline src/Component/Calculator.js:45]: - [ ] Please work on the functionality of the `%` currently its not working

File: src/Component/Calculator.js [hunk 2]
```diff
@@ -37,39 +35,38 @@ class Calculator extends React.Component {
         </div>
         <div className="calculator d-flex flex-column">
           <div className="calc-row d-flex align-content-stretch">
-            <button className="p-5" onClick={() => { this.eventHandler(calculate(this.state, 'AC')); }}>AC</button>
-            <button className="p-5" onClick={() => { this.eventHandler(calculate(this.state, '+/-')); }}>+/-</button>
-            <button className="p-5" onClick={() => { this.eventHandler(calculate(this.state, '%')); }}>%</button>
-            <button className="p-5 orange" onClick={() => { this.eventHandler(calculate(this.state, '÷')); }}>÷
+            <button className="p-5" onClick={() => { eventHandler(calculate(obj, 'AC')); }}>AC</button>
```
Comment (DeliceLydia) [inline src/Component/Calculator.js:43]: - [x] After doing an operation this button `AC` has to display a default `zero` in the screen please fix that

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
# Hi @Gracetemitope, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Component/Calculator.js (hunk 1 above)
Comment (Gracetemitope) [reply src/Component/Calculator.js:45]: Hi! The functionality works just perfectly from here. Can I know the error you having there?

**Comment by Gracetemitope** [comment]
Demo: https://www.loom.com/share/822d76bd233445d9a64f25a8bfa27ae0

File: src/Component/Calculator.js [hunk 3]
```diff
@@ -1,75 +1,76 @@
...
-    }
+  const obj = {
+    total: stotal,
+    next: snext,
+    operation: soperation,
+  };
+  const eventHandler = (newState) => {
+    // if (newState.total != null) {
+    //   setTotal(newState.total);
+    // }
```
Comment (MahmoudBakr23) [inline src/Component/Calculator.js:20]: - [ ] **OPTIONAL**
Kindly remove any commented-out code from your files. This is best practice for cleaner and more optimized code 😄 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hi @Gracetemitope 😃, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

### **Highlights**

- Thank you for recording a short video showing your project. 😄 

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [ ] Kindly fix the broken MIT and issues page links in the README file for a completely professional repo 🙏 

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

-----
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**